app/controllers/base_assets_controller.rb

```diff
@@ -22,6 +22,13 @@ def exclude_blank_redirect_url(params)
     params.reject { |k, v| (k.to_sym == :redirect_url) && v.blank? }
   end
 
+  def handle_empty_access_limited_param(params)
+    if params.has_key?(:asset) && params[:asset].has_key?(:access_limited) && params[:asset][:access_limited].empty?
+      params[:asset][:access_limited] = []
+    end
+    params
+  end
+
```

Comment:
This seems quite "ugly", but I can't see a way to simplify it. I think it would be useful to mention other solutions you considered, e.g. having two fields: `Asset#access_limit?` & `Asset#authorized_users`, and ideally capturing these ideas in a GitHub issue.

Reply:
I've updated the commit note in the initial commit to explain the alternative approach of having two fields.

I've captured the idea of using two fields in https://github.com/alphagov/asset-manager/issues/511.